### PR TITLE
Cleaning up code

### PR DESCRIPTION
**Pull request by joefol**

### CODE REVIEW

**Comment by joefol** [comment]
I think there might be an issue with the MultipinConfiguration class. All components using it (RGB, rotary encoder) get ambiguous dependency warnings.

**Comment by joefol** [comment]
Also there are no log messages implemented for the RFID scanner or for some methods in some of the other components.

**Review by ruthvikm — CHANGES_REQUESTED** [review]
I can see few lambda expression exceptions. Please fix them.
image

**Review by ruthvikm — COMMENTED** [review]
The exception "Ambiguous dependency: there are multiple beans that match the injection point" occurs due to the use of MuiltiPinConfiguration.java class that being a hub for PWMMultiPin and DigitalInputMultipin configs which are required for rgb led and rotary-encoder respectively.  The Dependency Injection gets confused here between these two beans. This expection can be used to create a new issue.

**Review by SwindleA — CHANGES_REQUESTED** [review]
Looks good. I think we should build and run the project before merging. We can do this during the next meeting.

File: components/src/main/java/com/opensourcewithslu/components/controllers/rgbController.java [hunk 1]
```diff
@@ -13,6 +12,7 @@
 @Controller("/rgb")
 public class rgbController {
     private final RGBLEDHelper rgbledHelper;
+
     private static final Logger log = LoggerFactory.getLogger(rgbController.class);
```
Comment (ruthvikm) [inline components/src/main/java/com/opensourcewithslu/components/controllers/rgbController.java:16]: Get rid of this, we didn't write any log messages here.

File: micronautpi4j-utils/src/main/java/com/opensourcewithslu/outputdevices/RGBLEDHelper.java [hunk 2]
```diff
@@ -60,11 +59,12 @@ public void setColorHex(String hex)
         green.on(g, 200);
         blue.on(b, 200);
     }
+
     //tag::method[]
     public void setColorHex(String hex, int[] frequency)
     //end::method[]
     {
-        log.trace("setting the color and frequency via hex and int");
+        log.info("setting the color and frequency via hex and int");
```
Comment (ruthvikm) [inline micronautpi4j-utils/src/main/java/com/opensourcewithslu/outputdevices/RGBLEDHelper.java:67]: Change this log message in RGBHelper class. Here it is "Setting the color via Hex and a list of frequencies(RGB)"

File: micronautpi4j-utils/src/main/java/com/opensourcewithslu/outputdevices/RGBLEDHelper.java [hunk 3]
```diff
@@ -79,54 +79,55 @@ public void setColorHex(String hex, int[] frequency)
...
+        log.info("set green");
         this.green.on(green, 200);
     }
 
     //tag::method[]
     public void setGreen(int green, int frequency)
     //end::method[]
     {
-        log.trace("set green and set frequency");
+        log.info("set green and set frequency");
```
Comment (ruthvikm) [inline micronautpi4j-utils/src/main/java/com/opensourcewithslu/outputdevices/RGBLEDHelper.java:122]: Change this to "Setting Green color and it's frequency"

File: micronautpi4j-utils/src/main/java/com/opensourcewithslu/outputdevices/RGBLEDHelper.java [hunk 4]
```diff
@@ -79,54 +79,55 @@ public void setColorHex(String hex, int[] frequency)
...
         this.green.on(green, frequency);
     }
 
     //tag::method[]
     public void ledOff()
     //end::method[]
     {
+        log.info("turning off each LED pin");
```
Comment (ruthvikm) [inline micronautpi4j-utils/src/main/java/com/opensourcewithslu/outputdevices/RGBLEDHelper.java:130]: Change the log message to "Turning off RGB LED"

**Review by ruthvikm — CHANGES_REQUESTED** [review]

**Comment by joefol** [comment]
> Looks good. I think we should build and run the project before merging. We can do this during the next meeting.

It builds successfully on my machine, but I haven't ran it on a pi.

**Review by ruthvikm — COMMENTED** [review]
Looks good. Once we test this on pi, we can merge this pull request.

File: components/src/main/java/com/opensourcewithslu/components/controllers/RotaryEncoderController.java [hunk 5]
```diff
@@ -9,11 +9,12 @@
...
 @Controller("/rotaryEncoder")
 public class RotaryEncoderController {
-
     private final RotaryEncoderHelper encoderHelper;
+
     private final RotaryEncoderHelper encoderHelper2;
 
-    public RotaryEncoderController(@Named("rotary-encoder")MultipinConfiguration rotaryEncoder, @Named("rotary-encoder-2")MultipinConfiguration rotaryEncoder2){
+    public RotaryEncoderController(@Named("rotary-encoder")MultipinConfiguration rotaryEncoder,
+                                   @Named("rotary-encoder-2")MultipinConfiguration rotaryEncoder2){
```
Comment (ruthvikm) [inline components/src/main/java/com/opensourcewithslu/components/controllers/RotaryEncoderController.java:17]: We didn't use this rotary-encoder-2, I think we can get rid off it.

**Review by ruthvikm — CHANGES_REQUESTED** [review]

**Review by ruthvikm — APPROVED** [review]
Tested most of the components. Everything works fine

**Review by SwindleA — APPROVED** [review]
Looks good.

**Pull request merged**